### PR TITLE
Add close/open commands

### PR DESCRIPTION
**Pull request by bkeepers**
slack_-_github_slack_dev

- [x] `/github close`
- [x] `/github reopen`
- [x] Tests
- [x] Put behind `EARLY_ACCESS` check
- [x] Allow `owner/repo#123` shorthand

cc #474

### CODE REVIEW

File: lib/commands/index.js [hunk 1]
```diff
@@ -26,12 +27,30 @@ module.exports = (robot) => {
...
     middleware.pendingCommand.clear,
     subscribe,
   );
 
+  app.post(
+    '/close',
+    middleware.earlyAccess,
+    middleware.authenticate,
+    middleware.getResource(),
+    middleware.getInstallation,
```
Comment (wilhelmklopp) [inline lib/commands/index.js]: Do we need `getInstallation`?

File: lib/commands/index.js [hunk 2]
```diff
@@ -26,12 +27,30 @@ module.exports = (robot) => {
...
+    issueState.bind(null, 'closed'),
+  );
+
+  app.post(
+    '/reopen',
+    middleware.earlyAccess,
+    middleware.authenticate,
+    middleware.getResource(),
+    middleware.getInstallation,
```
Comment (wilhelmklopp) [inline lib/commands/index.js]: same as above

File: lib/middleware/early-access.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+const hasEarlyAccess = require('../slack/has-early-access');
+
+module.exports = (req, res, next) => {
+  const params = { channelId: req.body.channel_id, teamId: req.body.team_id };
+  if (hasEarlyAccess(params)) {
+    req.log(params, 'Granting access to early access feature');
```
Comment (wilhelmklopp) [inline lib/middleware/early-access.js]: Should we specify a log level for these?

File: test/integration/issue-state.test.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    process.env.EARLY_ACCESS_CHANNELS = 'C2147483705';
+
+    user = await GitHubUser.create({
+      id: 2,
+      accessToken: 'github-token',
+    });
+
+    workspace = await SlackWorkspace.create({
+      slackId: 'T0001',
+      accessToken: 'xoxp-token',
```
Comment (wilhelmklopp) [inline test/integration/issue-state.test.js]: suuuuuuuper nit: `xoxa-token`

File: lib/middleware/early-access.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+const hasEarlyAccess = require('../slack/has-early-access');
+
+module.exports = (req, res, next) => {
```
Comment (wilhelmklopp) [inline lib/middleware/early-access.js]: wondering if it makes sense to also set `hasEarlyAccess` on `res.locals`?

File: lib/commands/index.js [hunk 6]
```diff
@@ -26,12 +27,30 @@ module.exports = (robot) => {
...
+    issueState.bind(null, 'open'),
```
Comment (wilhelmklopp) [inline lib/commands/index.js]: This is super neat, but might be confusing to folks reading it for the first time? Maybe a comment or a link explaining what this does would be good?

File: lib/commands/index.js (hunk 1 above)
Comment (bkeepers) [reply lib/commands/index.js]: Yes, I believe the user won't be able to take action without an installation.

**Comment by bkeepers** [comment]
This is dark shipped in production now. We'll add it to help and enable it for everyone soon.

**Pull request merged**